### PR TITLE
Add password authentication support for SSH

### PR DESCRIPTION
**Pull request by shaunluedeke**
Introduce optional password support in the Ssh class through the constructor and a new usePassword method. Updated README to document these changes and added relevant test cases.
#100

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "spatie/ssh",
+    "name": "shaunluedeke/ssh",
```
Comment (Nielsvanpach) [inline composer.json:2]: 😁

File: tests/SshTest.php [hunk 2]
```diff
@@ -164,3 +164,10 @@
 
     assertMatchesSnapshot($command);
 });
+
+it('can login with a password', function () {
```
Comment (Nielsvanpach) [inline tests/SshTest.php:168]: could you also provide a fail scenario with an invalid password?

File: composer.json (hunk 1 above)
Comment (shaunluedeke) [reply composer.json]: Sry I did not want to push with was for testing

File: tests/SshTest.php (hunk 2 above)
Comment (shaunluedeke) [reply tests/SshTest.php:168]: sorry i'm not at home right now i couldn't test it right now

**Comment by freekmurze** [comment]
Thank you!

**Pull request merged**